mmv1/products/gkeonprem/BareMetalCluster.yaml

```diff
@@ -0,0 +1,1003 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+--- !ruby/object:Api::Resource
+name: "BareMetalCluster"
+min_version: beta
+base_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters"
+create_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters?bare_metal_cluster_id={{name}}"
+update_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+delete_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}?force=true"
+self_link: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+update_verb: :PATCH
+update_mask: true
+description: "A Google Bare Metal User Cluster."
+autogen_async: false
+id_format: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+import_format: ["projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"]
+taint_resource_on_failed_create: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_basic"
+    min_version: beta
+    primary_resource_id: "cluster-basic"
+    vars:
+      name: "basic"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_manuallb"
+    min_version: beta
+    primary_resource_id: "cluster-manuallb"
+    vars:
+      name: "manuallb"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_bgplb"
+    min_version: beta
+    primary_resource_id: "cluster-bgplb"
+    vars:
+      name: "bgplb"
+    test_env_vars:
+      project: "fake-backend-360322"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "name"
+    description: The bare metal cluster name.
+    immutable: true
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: "location"
+    description: The location of the resource.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "adminClusterMembership"
+    required: true
+    immutable: true
```

Comment:
Could you confirm this is immutable? Looks like it's not mentioned in the doc https://cloud.devsite.corp.google.com/anthos/clusters/docs/on-prem-api/reference/rest/v1/projects.locations.bareMetalClusters.

Reply:
Thanks for checking. Yes, the adminClusterMembership is immutable currently. We do have the plan to make it mutable in the future. I will remove the `immutable: true` when it becomes mutable later.